app/monitoring/CloudWatch.scala

```diff
@@ -0,0 +1,65 @@
+package monitoring
+
+import java.util.concurrent.Future
+
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.regions.Regions.EU_WEST_1
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsyncClient
+import com.amazonaws.services.cloudwatch.model.{Dimension, MetricDatum, PutMetricDataRequest, PutMetricDataResult}
+import com.gu.aws.CredentialsProvider
+import monitoring.CloudWatch.cloudwatch
+
+trait CloudWatch extends TagAwareLogger {
+  val stage: String
+  val application : String
+  lazy val stageDimension: Dimension = new Dimension().withName("Stage").withValue(stage)
```

Comment:
if you used an abstract class with parameters or [delayed trait initialisation](https://stackoverflow.com/questions/16251379/delaying-trait-initialization) with respect to `stage` and `application` you could omit the `lazy val`, in which case `mandatoryDimensions` could be a regular `val` as well.